src/hotspot/cpu/aarch64/vm_version_aarch64.cpp

```diff
@@ -444,7 +444,19 @@ void VM_Version::initialize() {
   }
 
   if (UseSVE > 0) {
-    _initial_sve_vector_length = get_current_sve_vector_length();
+    int vl = get_current_sve_vector_length();
+    if (vl < 0) {
+      warning("Unable to get SVE vector length on this system. "
+              "Disabling SVE. Specify -XX:UseSVE=0 to shun this warning.");
+      FLAG_SET_DEFAULT(UseSVE, 0);
+    } else if (vl < FloatRegister::sve_vl_min || ((vl % FloatRegister::sve_vl_min) != 0) || !is_power_of_2(vl)) {
```

Comment:
I think `vl < FloatRegister::sve_vl_min` is redundant.
It is covered by `vl % FloatRegister::sve_vl_min) != 0`.

Reply:
I thought so too, but it is not, if `vl == 0`?

Reply:
It covers the case: `vl == 0`.

Reply:
According to https://man7.org/linux/man-pages/man2/prctl.2.html, `prctl` returns a nonnegative value. So we need to check `vl` for zero.

Reply:
How? `0 < 16` => `TRUE`, but `(0 % 16) != 0` => `FALSE`.
What I can see is that `0` is covered by `is_power_of_2`, however.

Reply:
Yes, our implementation of `is_power_of_2` returns `false` for 0.